### PR TITLE
[libc] added grouping of guarded functions in header generation

### PR DESCRIPTION
**Pull request by RoseZhang03**
Instead of #ifdef guards for each individual function, #ifdef and #endif
will surround all functions that have the same guard.

### CODE REVIEW

**Review by aaryanshukla — APPROVED** [review]
if this has been tested on guarded files then LGTM!

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Review by michaelrj-google — COMMENTED** [review]
This code looks good, but you should add tests to check that it works before landing it

**Review by michaelrj-google — APPROVED** [review]

**Pull request merged**
